src/main/java/com/sda/zdjavapol107/checkers/CheckersSave.java

```diff
@@ -0,0 +1,82 @@
+package com.sda.zdjavapol107.checkers;
+
+import com.sda.zdjavapol107.HibernateFactory;
+import com.sda.zdjavapol107.checkers.dao.*;
+import com.sda.zdjavapol107.checkers.model.*;
+import org.hibernate.SessionFactory;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+public class CheckersSave {
+
+    public void saveGame(CheckersBoard checkersBoard) {
+        Tournament tournament = convertTournamentToEntity(checkersBoard.getCheckersTournament());
+        Player playerOne = convertPlayerToEntity(checkersBoard.getPlayerOne());
+        Player playerTwo = convertPlayerToEntity(checkersBoard.getPlayerTwo());
+        playerOne.setTournament(tournament);
+        playerTwo.setTournament(tournament);
+        Game game = createGameEntity(playerOne, playerTwo, tournament);
+        Map<Integer, GamePeaces> gamePeacesMap = createGamePeacesMap(checkersBoard, game, playerOne, playerTwo);
+        Set<Move> setOfMoves = createSetOfMoves(checkersBoard, gamePeacesMap, playerOne, playerTwo, game);
+
+        SessionFactory sessionFactory = new HibernateFactory().getSessionFactory();
+        PlayerDao playerDao = new PlayerDao(sessionFactory);
+        GameDao gameDao = new GameDao(sessionFactory);
+        GamePeacesDao gamePeacesDao = new GamePeacesDao(sessionFactory);
+        MoveDao moveDao = new MoveDao(sessionFactory);
+        TournamentDao tournamentDao = new TournamentDao(sessionFactory);
+
+        tournamentDao.save(tournament);
+        playerDao.save(playerOne);
+        playerDao.save(playerTwo);
+        gameDao.save(game);
+        gamePeacesMap.values().forEach(gamePeacesDao::save);
+        setOfMoves.forEach(moveDao::save);
+    }
+
+    private Tournament convertTournamentToEntity(CheckersTournament checkersTournament) {
+        return new Tournament(checkersTournament.getPlayerOneScore(), checkersTournament.getPlayerTwoScore());
+    }
+
+    private Player convertPlayerToEntity(CheckersPlayer checkersPlayer) {
+        return new Player(checkersPlayer.getPlayerName());
+    }
+
+    private Game createGameEntity(Player playerOne, Player playerTwo, Tournament tournament) {
+        Game game = new Game(Set.of(playerOne, playerTwo), tournament);
+        game.setSaveDateAndTime(LocalDateTime.now());
+        return game;
+    }
+
+    private Map<Integer, GamePeaces> createGamePeacesMap(CheckersBoard checkersBoard, Game game, Player playerOne, Player playerTwo) {
```

Comment:
z tymi mapami wydaje mi sie jakies przekombinowane, ale na razie mozg mi paruje :P musze ogarnac jakas lepsza alternatywe albo przynajmniej pomysl jak to ladniej podizelic